src/operator/mshadow_op.h

```diff
@@ -84,41 +102,68 @@ struct negation {
 struct reciprocal {
   template<typename DType>
   MSHADOW_XINLINE static DType Map(DType a) {
-    return DType(1.0f/a);
+    float af(static_cast<float>(a));
```

Comment:
why?

Reply:
My idea was to do computations in float and cast back to DType in the end.

I see now this is not really the best. Here is what I will do:
- Whenever any math.h function is involved in an expression, I will do computations in
   float and cast to DType in the end
- I'll leave all other expressions (purely arithmetic, no math.h) as they are. This will of
   course result in garbage if DType is not floating point, but maybe this is intended